src/cli_util.rs

```diff
@@ -1416,6 +1418,22 @@ pub fn resolve_base_revs(
     }
 }
 
+fn add_multi_message_to_hint_if_mutliple_revisions(err: CommandError) -> CommandError {
+    if let CommandError::UserError { message, hint } = err {
+        let hint = if message.contains("more than one revision") {
+            let hint = hint.map(|h| format!("{h}\n")).unwrap_or_default();
+            Some(format!(
+                "{hint}If this was intentional, specify the `--multi` argument"
+            ))
+        } else {
+            hint
+        };
+        CommandError::UserError { message, hint }
```

Comment:
nit: I think you can match match `&err` instead of `err` in the `if let` above and then return the unmodified `err` in the else here. That way only the case where the hint changes results in a new `CommandError`.

Reply:
I was hoping the compiler will be smart enough to optimize it away. Matching on `&err` takes two extra clones, not that it matters here.

I'll change it to the "two extra clones" version unless somebody suggests a better idea.

Reply:
The compiler might be smart enough, but yes, it doesn't matter if we waste a microsecond in an error case. I think it just seems cleaner to not destructure and re-build it in the case where we don't change the hint.

Reply:
Actually, there's a much better way to do it with a `match`. I keep forgetting and reinventing it.